vignettes/adrs_pcwg3.Rmd

```diff
@@ -0,0 +1,454 @@
+  ) %>%
+  mutate(
+    AVALC = case_when(
+      AVALC == "PD" | BONE_AVALC == "PD" ~ "PD",      # Progressive Disease
+      AVALC == "CR" & BONE_AVALC == "NED" ~ "CR",     # Complete Response
+      AVALC == "PR" & BONE_AVALC %in% c("NON-PD", "PDu") ~ "PR",  # Partial Response
+      AVALC == "SD" & BONE_AVALC %in% c("NON-PD", "PDu", "NED") ~ "SD",  # Stable Disease
+      AVALC %in% c("NE") & BONE_AVALC %in% c("NE") ~ "NE",        # Not Evaluable
+      TRUE ~ NA_character_
+    ),
+    RSCAT="PCWG3 and RECIST 1.1 Modified",
+    RSTESTCD="OVRLRESPC",
+    RSTEST="Combined Overall Tumor Response by Investigator-Derived",
+    RSSTRESC=AVALC
+
+  ) %>%
+  derive_vars_merged_lookup(
+    dataset_add = param_lookup,
+    by_vars = exprs(RSCAT, RSTESTCD)  # Match on RSCAT and RSTESTCD
+  )
+
+```
+
+#Append  overall_tpr as rows into the primary analysis dataset adrs
+```{r, eval=TRUE, include=TRUE, message=FALSE}
+adrs <- bind_rows(
+  adrs %>% mutate(AVALC=RSSTRESC) %>%
+    derive_vars_merged_lookup(
+      dataset_add = param_lookup,
+      by_vars = exprs(RSCAT, RSTESTCD)  # Match on RSCAT and RSTESTCD
+    ),
+  overall_tpr %>% select(everything())  # Keep all variables
+)
+```
+
+### Derive `AVAL`
+
+Since the Prostate Cancer Working Group 3 (PCWG3) guidelines align closely with RECIST 1.1 categories for evaluating soft tissue, bone lesions, and overall disease progression, we can use the admiralonco::aval_resp() function to assign AVAL (numeric values ordered from best to worst response)
+
+
+```{r}
+adrs <- adrs %>%
+  mutate(
+    AVAL = aval_resp(AVALC)
+  )
+```
+
+```{r, echo=FALSE}
+dataset_vignette(
+  adrs %>%
+    arrange(!!!get_admiral_option("subject_keys"), PARAMN, RSSEQ),
+  display_vars = exprs(USUBJID, VISIT, RSCAT, RSTESTCD, RSEVAL, PARAMCD, PARAM, PARCAT1,AVALC,AVAL)
+)
+```
+
+### Partial Date Imputation and Deriving `ADT`, `ADTF`, `AVISIT`, `AVISITN` etc.
+
+If your data collection allows for partial dates, you could apply a company-specific imputation rule at this stage when deriving `ADT`. For this example, here we impute missing day to last possible date.
+
+```{r}
+adrs <- adrs %>%
+  derive_vars_dt(
+    dtc = RSDTC,
+    new_vars_prefix = "A",
+    highest_imputation = "D",
+    date_imputation = "last"
+  ) %>%
+  derive_vars_dy(
+    reference_date = TRTSDT,
+    source_vars = exprs(ADT)
+  ) %>%
+  derive_vars_dtm(
+    dtc = RSDTC,
+    new_vars_prefix = "A",
+    highest_imputation = "D",
+    date_imputation = "last",
+    flag_imputation = "time"
+  ) %>%
+  mutate(AVISIT = VISIT,
+         AVISITN=VISITNUM)
+```
+
+#Derive ASEQ
+```{r}
+
+adrs <- derive_var_obs_number(
+  dataset = adrs,
+  by_vars = exprs(STUDYID, USUBJID),
+  order = exprs(PARAMCD, AVISITN,ADT,AVALC),
+  new_var = ASEQ
+)
+
+```
+
+```{r, echo=FALSE}
+dataset_vignette(
+  adrs,
+  display_vars = exprs(USUBJID, PARAMCD, ASEQ, AVISIT,AVISITN,ADT)
+)
+```
+###Bone Lesion Confirmation (`The 2+2 flare Rule in PCWG3`)
```

Comment:
nothing is derived here for 2+2 , so we can add the information at the top to describe the logic